### PR TITLE
Escape string in git config user.name

### PR DESCRIPTION
**Pull request by FadyMak**
Issue #230

### CODE REVIEW

File: lib/git-user.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = function () {
     email = exec('git config --get user.email')
   } catch (e) {}
 
-  name = name && name.toString().trim()
+  name = name && JSON.stringify(name.toString().trim()).slice(1, -1)
```
Comment (zigomir) [inline lib/git-user.js]: How come `.slice` is needed here?

Comment (sayanriju) [inline lib/git-user.js]: The extra quotes seems to be a quirk on the node REPL. It should work fine if you are running it in a script.

Comment (FadyMak) [inline lib/git-user.js]: @sayanriju `JSON.stringify('hello')` will result in an additional set of quotes around the output:

```javascript
console.log(JSON.stringify('hey'))
// "hey"
```

i.e.: the resultant string is actually `"hey"` as opposed to just `hey`

In our case, we don't want the resultant Git `user.name` to be wrapped in quotes but we would like to escape them:

```javascript
var gitConfigName = 'hello "me" you'
var packageJSON = { author: JSON.stringify('hello "me" you').slice(1, -1) }
console.log(packageJSON)
// { author: 'hello \\"me\\" you' }
packageJSON = { author: JSON.stringify('hello "me" you') }
console.log(packageJSON)
// { author: '"hello \\"me\\" you"' }
```

I hope that helps clear it up a bit.

Comment (sayanriju) [inline lib/git-user.js]: My bad! I mean slash, not quote (The extra ones in \\"me\\"). :P
Anyway, it did clear things up.

BTW, I guess `JSON.stringify` is preferred over `String.replace` due to the slight performance hit?

**Comment by zigomir** [comment]
Thanks @FadyMak!

**Pull request merged**